compiler/src/main/java/org/qbicc/graph/literal/FunctionParameterLiteral.java

```diff
@@ -0,0 +1,46 @@
+package org.qbicc.graph.literal;
+
+import org.qbicc.graph.ValueVisitor;
+import org.qbicc.type.PointerType;
+import org.qbicc.type.ValueType;
+
+public class FunctionParameterLiteral extends Literal {
+    private final String name;
+    private final PointerType type;
```

Comment:
My only real complaint of this iteration is this type, which represents a pointer literal derived from a symbol, which is essentially exactly what `SymbolLiteral` is.  A secondary quibble is that this doesn't really represent a function parameter (assuming I understand the expected usage correctly), so it would have been better named `FunctionPointerLiteral` in which case its type should have been covariantly `PointerType`.

But either way I think it might be better to go directly to the more canonical `AddressOf(FunctionDeclarationHandle)` for cases where you need a `Value` that refers to a native function rather than introducing another variant of `SymbolLiteral`.

Reply:
Oh yes I can basically replace this whole thing with `SymbolLiteral` I see.

I do think a `Literal` solution makes more sense since at the LLVM level the function pointer must be passed in similarly to a constant. Since we want something like:
```
call void @JLT_start0(i8* (i8*)* @threadWrapper)
```
When I have tried treating the function pointer as a`Value` it is usually lowered treating the function pointer as though it can be assigned to a variable like this which doesn't compile:
```
%1 = i8* (i8*)* @threadWrapper
call void @JLT_start0(%1)
```

Reply:
> When I have tried treating the function pointer as a`Value` it is usually lowered treating the function pointer as though it can be assigned to a variable like this which doesn't compile:
> 
> ```
> %1 = i8* (i8*)* @threadWrapper
> call void @JLT_start0(%1)
> ```

Hmm that's definitely not what it is supposed to do.  I'll have a look in the LLVM backend to see what is going on there.

Reply:
This was my most recent hacky test to produce that result. See if thats what you had in mind.
https://github.com/theresa-m/qbicc/commit/f8aa10d8ee3b3a1660c6303f9513997cf3dfc593

Reply:
Thanks for your PR this morning. I have removed the FunctionParameterLiteral class and am passing threadWrapper as a pointer to a FunctionDeclaration now.